File: backend/MyDotNetWebsiteApi/Models/Enums.cs
// Enums

// Note: C# Projects don't need a "using" statement referencing this
// file since this project will find this project automatically 
// because this file is inside the C# project
// and they share a namespace

// TODO: Add these into potential entries for MediaTypes:
// public enum MediaType
// {
//     Book,
//     Movie,
//     TVShow,
//     VideoGame,
//     Comic,
//     Webcomic,
//     Website,
//     Article,
//     Image,
//     RadioShow,
//     Podcast,
//     TBD
// }

public enum VisibilityStatus
{
    Private,
    Public,
    Shared
}

// Site-Wide Permissions Levels for each User Account
public enum UserRoleLevel
{
    Basic,  // Basic access
    Moderator,  // Can approve, edit, delete any content
    Administrator  // Full Access
}

// Specific Roles for Specific MediaList etc.
public enum PermissionLevel
{
    Viewer,
    Editor,
    Manager
}

public enum CreatorRole
{
    Author,
    Director,
    Developer,
    Publisher,
    TBD
}

// Determines a MediaList's special behavior: protection from deletion, mutual exclusivity, and site-wide visibility
public enum MediaListCategory
{
    Standard = 0,       // Regular user list — can be freely created and deleted
    VisitingStatus = 1, // One of the per-user mutually exclusive status lists (e.g. "Want to Visit", "Currently Visiting")
    Library = 2,        // Per-user protected singleton — remembers items without implying visit intent (like Spotify's "Liked Songs")
    Featured = 3,       // Admin-owned, site-wide, always Public — contents may be surfaced in special site sections
}

File: frontend/src/types/enums.ts
// TODO: NOTE: I just copy-pasted this enum from the backend/MyDotNetWebsiteApi/Models/Enums.cs file
// and translated it to .ts (aka TypeScript):


export enum VisibilityStatus {
    Private = 0,
    Public = 1,
    Shared = 2
}

// Mirror of backend MediaListCategory enum — determines list behavior, badge display, and deletion rules
// NOTE: Uses string values to match what the backend's JsonStringEnumConverter serializes (e.g. "VisitingStatus", not 1)
export enum MediaListCategory {
    Standard = 'Standard',             // Regular user list — can be freely created and deleted
    VisitingStatus = 'VisitingStatus', // One of the per-user mutually exclusive status lists (e.g. "Want to Visit")
    Library = 'Library',               // Per-user protected singleton — remembers items without implying visit intent
    Featured = 'Featured',             // Admin-owned, site-wide — contents may be surfaced in special site sections
}